README.md

```diff
@@ -1,36 +1,121 @@
+  Nethermind (dev)
+- A
+  [Zig implementation](https://github.com/keep-starknet-strange/ziggy-starkdust),
+  by Community (dev)
+- A [C++ implementation](https://github.com/lambdaclass/cairo-vm.c) by Lambda
+  Class (dev)
+
+The Lambda Class alt-implementations comes with a detailed guide
+([Go](https://github.com/lambdaclass/cairo-vm_in_go/blob/main/README.md#documentation),
+[C++](https://github.com/lambdaclass/cairo-vm.c?tab=readme-ov-file#documentation))
+on how they built their Cairo VM. It gives insights into the overall Cairo VM
+but is incomplete and rather specific to language details.
+
+Why would you have different implementations of the same program in multiple
+languages? For **implementation diversity**. As more implementations provide
+more:
 
-<!-- TODO: add some context about the need of client diversity -->
+- **Resilience**. It helps in finding bugs in the existing   implementations.
+- **Documentation**. The documentation over the Cairo VM is   still scarce, and
+  the latest version in prod (Rust) is not easy to read for   the average dev.
+- **Architecture diversity**. Different architectures can be   implemented to
+  achieve the same goal (e.g. memory model). However, most of the current
+  implementations essentially are a rewrite of the Rust implementation, which is
+  an (enhanced) rewrite of the Python implementation itself.
+- **Usage diversity**. The primary goals of each implementation can differ. For
+  example, the primary goals of production implementations are performance and
+  safety (Rust).
 
-The Go implementation comes with an
-[impressive documentation](https://github.com/lambdaclass/cairo-vm_in_go/blob/main/README.md#documentation)
-regarding the how-to for a Cairo VM. This repo is coded with the flow while
-reading it.
+The primary goal of our TypeScript implementation is **education** through
+**readability** and **expressiveness**.
 
-Read so far up to
-[#Run Context](https://github.com/lambdaclass/cairo-vm_in_go/blob/main/README.md#runcontext)
+### Demistifying the Cairo VM
 
-## Requirements
+The Cairo VM is still obscure to most of us, from a lack of documentation and
+hard-to-read languages obfuscating the inner workings of the VM. Nobody wants to
+struggle to reconstruct the missing pieces to understand how something works.
+
+We are implementing a Cairo VM in TypeScript to provide a well documented and
+easy-to-comprehend Cairo VM
+
+- TypeScript is easily readable and known by most devs if not all
+- Deliberate design choices to further improve readability and simplicity
+- Extensive documentation: JSDoc, diagrams, explainers, etc.
+
+## State of the VM
+
+| Goals                        | Done? |
+| ---------------------------- | ----- |
+| Run basic Cairo Zero program | 🗹     |
+| Run basic Cairo program      | ☐     |
+| Run any Cairo Zero program   | ☐     |
+| Run any Cairo program        | ☐     |
+| Run any StarkNet contract    | ☐     |
+| Benchmark against other VMs  | ☐     |
```

Comment:
can't you talk instead in terms of concrete steps like simple programs, builtins, hints, starknet ?